akka-docs/rst/java/stream/stages-overview.rst

```diff
@@ -959,6 +959,89 @@ merging. The maximum number of merged sources has to be specified.
 **completes** when upstream completes and all consumed substreams complete
 
 
+Time aware stages
+-----------------
+
+Those stages operate taking time into consideration.
+
+initialTimeout
+^^^^^^^^^^^^^^
+If the first element has not passed through this stage before the provided timeout, the stream is failed
+with a ``TimeoutException``.
+
+**emits** when upstream emits an element
+
+**backpressures** when downstream backpressures
+
+**completes** when upstream completes or fails if timeout elapses before first element arrives
+
+**cancels** when downstream cancels
+
+completionTimeout
+^^^^^^^^^^^^^^^^^
+If the completion of the stream does not happen until the provided timeout, the stream is failed
+with a ``TimeoutException``.
+
+**emits** when upstream emits an element
+
+**backpressures** when downstream backpressures
+
+**completes** when upstream completes or fails if timeout elapses before upstream completes
+
+**cancels** when downstream cancels
+
+idleTimeout
+^^^^^^^^^^^
+If the time between two processed elements exceeds the provided timeout, the stream is failed
+with a ``TimeoutException``.
+
+**emits** when upstream emits an element
+
+**backpressures** when downstream backpressures
+
+**completes** when upstream completes or fails if timeout elapses between two emitted elements
+
+**cancels** when downstream cancels
+
+backpressureTimeout
+^^^^^^^^^^^^^^^^^^^
+If the time between the emission of an element and the following downstream demand exceeds the provided timeout,
+the stream is failed with a ``TimeoutException``. If the producer doesn't emmit any element,
```

Comment:
I don't get this: "If the producer doesn't emmit any element, the timeout is still valid and will fail the stream". This basically then describes `idleTimeout` which is not what I intended with this stage.

The semantics I was after:
 - it is an error to block the upstream producer for a too long time (time between push->onPull must be bounded)
 - it is perfectly fine for the upstream producer to not produce any elements for a long time (time between pull->onPush can be unbounded)

I check the impl now, it might be just the description here is misleading.